tests/test_xyz/test_polygon.py

```diff
@@ -288,31 +373,168 @@ def test_check_column_names():
+            "Z_TVDSS": [3.0, 3.0],
+            "POLY_ID": [3, 3],
+            "T_DELTALEN": [2.0, 1.0],
+        }
+    )
     pol.dataframe = data
+    pol._pname = "POLY_ID"
+    pol._tname = "T_DELTALEN"
 
     with pytest.raises(ValueError, match="Wrong type of input"):
-        setattr(pol, name_attribute, test_name)
+        setattr(pol, "tname", test_name)
 
 
 @pytest.mark.parametrize(
-    "name_attribute", [("hname"), ("dhname"), ("tname"), ("dtname"), ("pname")]
+    "name_attribute", [("hname"), ("dhname"), ("tname"), ("dtname")]
 )
-def test_raise_non_existing_name(name_attribute):
+def test_raise_special_name_name_type(name_attribute):
     pol = Polygons()
-    data = pd.DataFrame({"ANYTHING": [1, 2]})
+    data = pd.DataFrame(
+        {
+            "X_UTME": [1.0, 2.0],
+            "Y_UTMN": [2.0, 1.0],
+            "Z_TVDSS": [3.0, 3.0],
+            "POLY_ID": [3, 3],
+            "T_DELTALEN": [2.0, 1.0],
+        }
+    )
+    pol._pname = "POLY_ID"
     pol.dataframe = data
 
-    with pytest.raises(ValueError, match="does not exist"):
-        setattr(pol, name_attribute, "NON_EXISTING")
+    with pytest.raises(ValueError, match="does not exist as a column"):
+        setattr(pol, name_attribute, "anyname")
+
+
+@pytest.mark.parametrize(
+    "func, where, value, expected_result",
+    [
+        ("add", "inside", 44.0, [47.0, 47.0, 3.0]),
+        ("add", "outside", 44.0, [3.0, 3.0, 47.0]),
+        ("mul", "inside", 2.0, [6.0, 6.0, 3.0]),
+        ("div", "inside", 2.0, [1.5, 1.5, 3.0]),
+        ("sub", "outside", 3.0, [3.0, 3.0, 0.0]),
+        ("set", "outside", 88.0, [3.0, 3.0, 88.0]),
+        ("eli", "outside", 0, [3.0, 3.0]),
+    ],
+)
+@pytest.mark.parametrize("shorthand", [True, False])
+@pytest.mark.parametrize("constructor", [Points, Polygons])
+def test_polygons_operation_in_polygons(
+    func, where, value, expected_result, shorthand, constructor
+):
+    """Test what happens to point belonging to polygons for add_inside etc."""
+
+    closed_poly = Polygons(
+        pd.DataFrame(
+            {
+                "X_UTME": [0.0, 100.0, 100.0, 0.0, 0.0],
+                "Y_UTMN": [0.0, 0.0, 200.0, 200.0, 0.0],
+                "Z_TVDSS": [3.0, 3.0, 3.0, 3.0, 3.0],
+                "POLY_ID": [0, 0, 0, 0, 0],
+            }
+        )
+    )
+
+    # this set will first point on border, middle point inside and last point outside
+    # the closed_poly defined above
+    pointset = constructor(
+        pd.DataFrame(
+            {
+                "X_UTME": [0.0, 50.0, 200.0],
+                "Y_UTMN": [0.0, 100.0, 300.0],
+                "Z_TVDSS": [3.0, 3.0, 3.0],
+                "POLY_ID": [0, 0, 0],
+            }
+        )
+    )
+
+    inside = True if "inside" in where else False
+
+    if shorthand:
+        if func == "eli":
+            getattr(pointset, f"eli_{where}")(closed_poly)
+        else:
+            getattr(pointset, f"{func}_{where}")(closed_poly, value)
+    else:
```

Comment:
Understand why you do this, but might make sense to have in a different test to avoid forking?

Reply:
So a different test for "eli" ? That can be arranged :)

Reply:
I started looking at this and trying to split this into two tests seemed to create a lot of noise in my opinion (more test data setup and copy-pasting parametrization). If you have any ideas about how to do this cleanly I'll try again.

Reply:
Fair enough, fine with leaving it as is 👍